doc/adrs/046_make_the_workbench_panels_content_configurable.adoc

```diff
@@ -0,0 +1,74 @@
+= ADR-046 - Make the workbench panels' content configurable
+
+== Context
+
+The `Workbench` frontend component is responsible for displaying and coordinating the different views and representations in the main project edition page.
+Currently its layout and content is hard-coded, with the _explorer_ and _validation view_ on top of each other in the left panel, and the _details view_ and _representations view_ on the right panel.
+For some applications, some of these views are not relevant or should be replaced by application-specific alternatives.
+
+== Decision
+
+The workbench will keep its overall organisation in 3 panels, but the left and right panels will no longer hard-code the list of views they contain.
+
+Instead the workbench will accept a new kind of children element, `WorkbenchViewContribution` to indicate the list of components to place in the left or right panels.
+
+```js
+<Workbench
+  editingContextId={project.currentEditingContext.id}
+  initialRepresentationSelected={representation}
+  onRepresentationSelected={onRepresentationSelected}
+  readOnly={project.accessLevel === 'READ'}>
+  <WorkbenchViewContribution side="left" title="Validation" component={ValidationWebSocketContainer} />
+  <WorkbenchViewContribution side="left" title="Explorer" component={ExplorerWebSocketContainer} />
+  <WorkbenchViewContribution side="right" title="Details" component={PropertiesWebSocketContainer} />
+  <WorkbenchViewContribution side="right" title="Representations" component={RepresentationsWebSocketContainer} />
+  <TreeItemContextMenuContribution canHandle={(item: TreeItemType) => item.kind.startsWith('obeoStudio://document')}
```

Comment:
`siriusWeb://` instead of `obeoStudio://`

Reply:
Done